apps/ex_nvr/lib/ex_nvr/model/device.ex

```diff
@@ -167,13 +167,210 @@ defmodule ExNVR.Model.Device do
     end
   end
 
+  defmodule SnapshotConfig do
```

Comment:
Consider moving this module to its own file, it seems too big to be included in the device file.